src/services/completions.ts

```diff
@@ -1495,9 +1501,20 @@ namespace ts.Completions {
                 // Add filtered items to the completion list
                 symbols = filterObjectMembersList(typeMembers, Debug.assertDefined(existingMembers));
             }
+            setSortTextToOptionalMember();
+
             return GlobalsSearch.Success;
         }
 
+        // Set SortText to OptionalMember if it is an optinoal member
+        function setSortTextToOptionalMember() {
+            symbols.forEach(m => {
+                if (SymbolDisplay.getSymbolModifiers(m) === "optional") {

```

Comment:
This check just needs to be `m.flags & SymbolFlags.Optional` instead of getting display string